src/Sensor.cc

```diff
@@ -20,9 +20,21 @@ double Sensor::sensorNormalOffset() const {
   return offset;
 }
 
+double Sensor::sensorXOffset() const {
+  double offset;
+  if (parent_->numSensors() <= 1) offset = 0.;
+  else {
+    if (myid() == 1) offset = -parent_->offsetForSensors();
+    else offset = parent_->offsetForSensors();
+  }
+  return offset;
+}
+
+
 const Polygon3d<4>& Sensor::hitPoly() const {
   double offset = sensorNormalOffset();
-  if (hitPoly_ == 0) hitPoly_ = CoordinateOperations::computeTranslatedPolygon(parent_->basePoly(), offset);
+  double xOffset = sensorXOffset();
+  if (hitPoly_ == 0) hitPoly_ = CoordinateOperations::computeTranslatedPolygon(parent_->basePoly(), offset, xOffset);
```

Comment:
The way it looks to me (from just looking at this function) 'compteTranslatedPolygon' should only be invoked if hitPoly_ is not set - so it's not called every time hitPoly() is called. That is, unless the previous hitPoly() call didn't set hitPoly_ - but it should fill it with something even if both offset and xOffset are 0, so in practice I don't think that would happen.

Also to note: this doesn't really change how hitPoly_ was assigned previously (it extends it), so it shouldn't have become more inefficient (hopefully :-) )

Reply:
Oh, yes. Now I remember! … I wrote it :-)
Yes, indeed, it's done so that it's computed on the fly, and only when it's needed, but only once.

Thanks for reminding me!